ls1intum/Artemis
# `Development`: Fix flaky StructuralTestCaseServiceTest

### Checklist
#### General
- [x] The changed server tests succeed **locally** and on **Bamboo** and on **GitHub Actions**.
#### Server

### Motivation and Context
Multiple tests in StructuralTestCaseServiceTest were flaky. The tests frequently failed because of a Files.walk method that failed with a `NoSuchFileException`. The file that could not be found is called gc.log.lock, indicating that _git gc_ is still enabled, even though disabled in the `GitService` class.

### Description
In GitService, there is already an attempt to disable _auto gc_:
`gitRepoConfig.setInt(ConfigConstants.CONFIG_GC_SECTION, null, ConfigConstants.CONFIG_KEY_AUTO, 0);` 
However, this is not enough and more options need to be set to prevent the automatic execution of `git gc`.
Multiple options where taken from a helper method in `ProgrammingExerciseTestService`, disabling auto gc.

MySQL run without failing StructuralTestCaseServiceTest(s):
https://github.com/ls1intum/Artemis/actions/runs/6504427907/job/17666361382?pr=7328

Postgres run without failing StructuralTestCaseServiceTest(s):
https://github.com/ls1intum/Artemis/actions/runs/6504427907/job/17666361159?pr=7328

### Steps for Testing
_Although the behaviour should not have changed. I'd still like two reviewers to test Programming Exercises._
1. Create Programming Exercises, make sure the template and solution build run as expected.
2. As a student, participate in a newly created Programming Exercise (locally and on a test server).
3. As a student, participate in a newly created Programming Exercise in an Exam.

### Review Progress

#### Code Review
#### Manual Tests
